xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -355,6 +355,9 @@ bool CGUIControlFactory::GetTexture(const TiXmlNode* pRootNode, const char* strT
   if (flipY && StringUtils::CompareNoCase(flipY, "true") == 0)
     image.orientation = 3 - image.orientation; // either 3 or 2
   image.diffuse = XMLUtils::GetAttribute(pNode, "diffuse");
+  const char* diffuseslice = pNode->Attribute("sliceonlydiffuse");
+  if (diffuseslice && StringUtils::CompareNoCase(diffuseslice, "true", 4) == 0)
```

Comment:
Why only compare the first 4 chars? No opposition just curious.

Reply:
because no case comparision do we typically allow `true | TRUE | or TrUe` ?

Otherwise you could just do the comparison `diffuseslice == "true"`